src/simulation.jl

```diff
@@ -553,11 +602,10 @@ gensim(collate(c); dimension=dimension)
 
 Generate a simulation function from the given Decapode. The returned function can then be combined with a mesh and a function describing function mappings to return a simulator to be passed to `solve`.
 """
-gensim(d::AbstractNamedDecapode; dimension::Int=2, stateeltype = Float64) = gensim(d,
-                                                                                   vcat(collect(infer_state_names(d)), d[incident(d, :Literal, :type), :name]), dimension=dimension, stateeltype=stateeltype)
+gensim(d::AbstractNamedDecapode; dimension::Int=2, stateeltype = Float64, code_target = cpu()) = gensim(d, vcat(collect(infer_state_names(d)), d[incident(d, :Literal, :type), :name]), dimension=dimension, stateeltype=stateeltype, code_target=code_target)
 
-evalsim(d::AbstractNamedDecapode; dimension::Int=2, stateeltype = Float64) = eval(gensim(d, dimension=dimension, stateeltype=stateeltype))
-evalsim(d::AbstractNamedDecapode, input_vars; dimension::Int=2, stateeltype = Float64) = eval(gensim(d, input_vars, dimension=dimension, stateeltype=stateeltype))
+evalsim(d::AbstractNamedDecapode; dimension::Int=2, stateeltype = Float64, code_target = cpu()) = eval(gensim(d, dimension=dimension, stateeltype=stateeltype, code_target=code_target))
```

Comment:
Too long

Reply:
Fixed for now, we should just pass all keywords in the coming compiler update.